### PR TITLE
FeverAPI 32-bit fixes

### PR DESCRIPTION
**Pull request by Alkarex**
https://github.com/FreshRSS/FreshRSS/issues/1962

### CODE REVIEW

**Comment by Alkarex** [comment]
@kevinpapst

**Review by Frenzie — APPROVED** [review]
Looks straightforward

File: p/api/fever.php [hunk 1]
```diff
@@ -69,11 +69,11 @@ public function findEntries(array $feed_ids, array $entry_ids, $max_id, $since_i
 		if (!empty($entry_ids)) {
 			$bindEntryIds = $this->bindParamArray('id', $entry_ids, $values);
 			$sql .= " id IN($bindEntryIds)";
-		} else if (!empty($max_id)) {
+		} elseif ($max_id !== null) {
 			$sql .= ' id < :id';
 			$values[':id'] = $max_id;
 			$order = ' ORDER BY id DESC';
-		} else {
+		} elseif ($since_id !== null) {
```
Comment (kevinpapst) [inline p/api/fever.php]: This was wrong before, so just the question if we want to make it fail safe:
Looks like the calling part makes sure that either since_id or max_id is not null. But if we consider that someone is trying to use the API programmatically, then we should move both the WHERE and the ORDER (or at least the WHERE) clause into the if-else. Otherwise the SQL would be broken if someone calls this function with max_id=null and since_id=null. 
Or probably use an array to collect all WHERE parts and append it with `implode(' AND ', $where)` before $order is added to $sql.

Comment (Alkarex) [reply p/api/fever.php]: Good point. Here is a suggestion https://github.com/FreshRSS/FreshRSS/pull/1964/commits/44d3e5172655d2d9be4ae5ac383371b251eeae62

File: p/api/fever.php [hunk 2]
```diff
@@ -471,17 +471,18 @@ protected function getItems()
...
+			if (!ctype_digit($max_id) || $max_id < 0) {
+				$max_id = null;
 			}
 		} else if (isset($_REQUEST['with_ids'])) {
 			$entry_ids = explode(',', $_REQUEST['with_ids']);
 		} else {
 			// use the since_id argument to request the next $item_limit items
-			$since_id = isset($_REQUEST['since_id']) && is_numeric($_REQUEST['since_id']) ? intval($_REQUEST['since_id']) : 0;
+			$since_id = '' . $_REQUEST['since_id'];
+			if (!ctype_digit($since_id) || $since_id < 0) {
```
Comment (kevinpapst) [inline p/api/fever.php]: If I read the docu correct, a negative sign would already not be accepted by `ctype_digit` as it is a non numeric value. So I guess the <0 check is not required.
Also from the documentation: the string 000000001 would be accepted, but I don't know what the database makes out of it in that case `id > 000000001`. Thats why I originally used intval().
But to be honest, I didn't test these edge cases.

**Comment by kevinpapst** [comment]
So you pushed while I was still reviewing and typing. LOL, seems you catched my remarks before I even submitted them ;-) 
Looks good to me.

File: p/api/fever.php (hunk 1 above)
Comment (kevinpapst) [reply p/api/fever.php]: The old 1=1 trick is also fine :-D
